src/transformers/models/textnet/configuration_textnet.py

```diff
@@ -0,0 +1,183 @@
+            The num of channels in output for list of conv in stage 1.Should be of same length os `stage1_in_channels`
+        stage1_kernel_size (`List[int]`, *optional*, defaults to `[[3, 3], [3, 3], [3, 3]]`):
+            The kernel sizes for list of conv in stage 1.Should be of same length os `stage1_in_channels`
+        stage1_stride (`List[int]`, *optional*, defaults to `[1, 2, 1]`):
+            The strides for list of conv in stage 1.Should be of same length os `stage1_in_channels`
+        stage2_in_channels (`List[int]`, *optional*, defaults to `[64, 128, 128, 128]`):
+            The num of channels in input for list of conv in stage 2.
+        stage2_out_channels (`List[int]`, *optional*, defaults to `[128, 128, 128, 128]`):
+            The num of channels in output for list of conv in stage 2.Should be of same length os `stage2_in_channels`
+        stage2_kernel_size (`List[List[int]]`, *optional*, defaults to `[[3, 3], [1, 3], [3, 3], [3, 1]]`):
+            The kernel sizes for list of conv in stage 2.Should be of same length os `stage2_in_channels`
+        stage2_stride (`List[int]`, *optional*, defaults to `[2, 1, 1, 1]`):
+            The strides for list of conv in stage 2.Should be of same length os `stage2_in_channels`
+        stage3_in_channels (`List[int]`, *optional*, defaults to `[128, 256, 256, 256]`):
+            The num of channels in input for list of conv in stage 3.
+        stage3_out_channels (`List[int]`, *optional*, defaults to `[256, 256, 256, 256]`):
+            The num of channels in output for list of conv in stage 3.Should be of same length os `stage3_in_channels`
+        stage3_kernel_size (`List[List[int]]`, *optional*, defaults to `[[3, 3], [3, 3], [3, 1], [1, 3]]`):
+            The kernel sizes for list of conv in stage 3.Should be of same length os `stage3_in_channels`
+        stage3_stride (`List[int]`, *optional*, defaults to `[2, 1, 1, 1]`):
+            The strides for list of conv in stage 3.Should be of same length os `stage3_in_channels`
+        stage4_in_channels (`List[int]`, *optional*, defaults to `[256, 512, 512, 512]`):
+            The strides for list of conv in stage 4.Should be of same length os `stage4_in_channels`
+        stage4_out_channels (`List[int]`, *optional*, defaults to `[512, 512, 512, 512]`):
+            The num of channels in output for list of conv in stage 4.Should be of same length os `stage4_in_channels`
+        stage4_kernel_size (`List[List[int]]`, *optional*, defaults to `[[3, 3], [3, 1], [1, 3], [3, 3]]`):
+            The kernel sizes for list of conv in stage 4.Should be of same length os `stage4_in_channels`
+        stage4_stride (`List[int]`, *optional*, defaults to `[2, 1, 1, 1]`):
+            The strides for list of conv in stage 4.Should be of same length os `stage4_in_channels`
+        hidden_sizes (`List[int]`, *optional*, defaults to `[64, 64, 128, 256, 512]`):
```

Comment:
I am not sure I am understanding this correctly. stage_x_in_channels param is a list for each stage. It could have any  sizes rights . For example, for stage4, we infer the last element of stage_x_out_channels from hidden_sizes, But cannot infer the earlier out_channels , similar case of stage_x_in_channels. Let me know if I am not understanding this correctly .